src/preview.ts

```diff
@@ -0,0 +1,182 @@
+import { getInput, setOutput } from "@actions/core";
+import * as exec from "@actions/exec";
+import * as github from "@actions/github";
+import { StainlessV0 as Stainless } from "stainless";
+import { checkResults, runBuilds } from "./build";
+import { isConfigChanged } from "./config";
+import { generatePreviewComment, upsertComment } from "./comment";
+
+async function getParentCommits() {
+  // Get HEAD and BASE shas
+  const HEAD = github.context.payload.pull_request!.head.sha;
+  const BASE = github.context.payload.pull_request!.base.sha;
```

Comment:
similarly to above, I vaguely feel like this should be an explicit input into the workflow for visibility reasons